events/api.py

```diff
@@ -2511,6 +2511,37 @@ def _terms_to_regex(terms, operator, fuzziness=3):
     return regex.compile(expr, regex.IGNORECASE)
 
 
+def _get_queryset_from_cache(params, param, cache_name, operator, queryset):
+    cache_values = cache.get(cache_name)
+    val = params.get(param, None)
+
+    if not val or not cache_values:
+        return queryset
+
+    rc = _terms_to_regex(val, operator)
+    ids = {k for k, v in cache_values.items() if rc.search(v, concurrent=True)}
+    queryset = queryset.filter(id__in=ids)
+
+    return queryset
```

Comment:
I think checking `val` first and then calling the cache if needed could be a nice micro optimization. Also I think it might be a good idea to maybe send a message to sentry if we miss a cache, since that basically means that some service might be getting a bad result? :thinking: Or at least we'd then have more info when the service admins start asking us why linked events is returning a bad result. Though calling Sentry might be too much if we can expect to get cache misses, but I have a feeling that we shouldn't expect that. 

```suggestion
def _get_queryset_from_cache(params, param, cache_name, operator, queryset):
    val = params.get(param, None)
    if not val:
        return queryset
    
    cache_values = cache.get(cache_name)
    
    if not cache_values:
        send_a_message_to_sentry(f"Missed cache {cache_name}")  # Can't remember what function to actually call from the top of my head. :(
        return queryset

    rc = _terms_to_regex(val, operator)
    ids = {k for k, v in cache_values.items() if rc.search(v, concurrent=True)}
    queryset = queryset.filter(id__in=ids)

    return queryset
```

val = params.get(param, None)

Reply:
Same applies for the `_get_queryset_from_cache_many`

Reply:
Well, I guess the following should go nicely to sentry:

```python
import logging
logger = logging.getLogger(__name__)

...

logger.error(f"Missed cache {cache_name}")

```

Reply:
Changed as suggested.